src/workers/parser/mapOriginalExpression.js

```diff
@@ -81,22 +77,74 @@ export default function mapOriginalExpression(
     }
   }
 
-  if (replacements.size === 0) {
-    // Avoid the extra code generation work and also avoid potentially
-    // reformatting the user's code unnecessarily.
-    return expression;
+  return replacements;
+}
+
+// input `var a = 3` => `if (!window.hasOwnProperty("a")) window.a = 3 `
+function globalizeAssignment(node) {
+  return node.declarations.map(({ id, init }) => {
+    if (init) {
+      return t.ifStatement(
+        t.unaryExpression(
+          "!",
+          t.callExpression(
+            t.memberExpression(
+              t.identifier("self"),
+              t.identifier("hasOwnProperty")
+            ),
+            [t.stringLiteral(id.name)]
```

Comment:
This is going to fail if users are using destructuring since `id` is a general pattern.